types/index.d.ts

```diff
@@ -0,0 +1,461 @@
+// TypeScript Version: 3.5
+import * as React from 'react';
+import * as Client from 'stream-chat';
+import SeamlessImmutable from 'seamless-immutable';
+
+export interface ChatProps {
+  client: Client.StreamChat;
+  theme?: Theme;
+}
+
+export type Theme = 'messaging light' | 'messaging dark';
+
+interface ChannelProps {
+  /** Which channel to connect to, will initialize the channel if it's not initialized yet */
+  channel?: Client.Channel;
+  /** Client is passed automatically via the Chat Context */
+  client?: Client.StreamChat;
+  /** The loading indicator to use */
+  LoadingIndicator?: React.ElementType;
+  Message?: React.ElementType;
```

Comment:
@vishalnarkhede are you sure `React.ElementType` is the right type? in the component ref (different type system ofc) we say that it can be `func` or `node`

Reply:
ideally we'd include some more typing information for props of Message, Attachment, User, .. props. This would improve usability a lot

Reply:
@tbarbugli `React.ElementType` covers all the cases - node, functional component, class components etc etc. But yeah, more typing information will be useful here. I will add that :)